tools/github-team-user-store/GitHubTeamUserStore/GitHubTeamUserStore/GitHubEventClient.cs

```diff
@@ -0,0 +1,201 @@
+                // azure-sdk-write. If a ForbiddenException is encountered, then report it and return an
+                // empty list.
+                catch (Octokit.ForbiddenException forbiddenEx)
+                {
+                    Console.WriteLine($"{team.Name} cannot be retrieved using a GitHub PAT.");
+                    Console.WriteLine(forbiddenEx.Message);
+                    return emptyUserList.AsReadOnly();
+                }
+                // The only time we should get down here is if there's an exception caused by a network hiccup.
+                // Sleep for a second and try again.
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.Message);
+                    Console.WriteLine($"delaying {_delayTimeInMs} and retrying");
+                    await Task.Delay(_delayTimeInMs);
+                }
+            }
+            throw new ApplicationException($"Unable to get members for team {team.Name}. See above exception(s)");
+        }
+
+        /// <summary>
+        /// Given an Octokit.Team, call to get all child teams.
+        /// </summary>
+        /// <param name="team">Octokit.Team, the team whose child teams to retrieve.</param>
+        /// <returns>IReadOnlyList of Octkit.Team</returns>
+        /// <exception cref="ApplicationException">Thrown if GetAllChildTeams fails after all retries have been exhausted</exception>
+        public async Task<IReadOnlyList<Team>> GetAllChildTeams(Team team)
+        {
+            int tryNumber = 0;
+            while (tryNumber < _numRetries)
+            {
+                tryNumber++;
+                try
+                {
+                    return await _gitHubClient.Organization.Team.GetAllChildTeams(team.Id, _apiOptions);
+                }
+                // The only time we should get down here is if there's an exception caused by a network hiccup.
+                // Sleep for a second and try again.
+                catch (Exception ex)
+                {
+                    Console.WriteLine(ex.Message);
+                    Console.WriteLine($"delaying {_delayTimeInMs} and retrying");
+                    await Task.Delay(_delayTimeInMs);
+                }
+            }
+            throw new ApplicationException($"Unable to get members for team {team.Name}. See above exception(s)");
+        }
+
+        /// <summary>
+        /// Store the team/user blob in Azure blob storage.
+        /// </summary>
+        /// <param name="rawJson">The json string, representing the team/user information, that will be uploaded to blob storage.</param>
+        /// <returns></returns>
+        /// <exception cref="ApplicationException">If there is no AZURE_SDK_TEAM_USER_STORE_SAS in the environment</exception>
+        public async Task UploadToBlobStorage(string rawJson)
+        {
+            var azureSdkWriteTeamsSAS = Environment.GetEnvironmentVariable("AZURE_SDK_TEAM_USER_STORE_SAS");
```

Comment:
Similar to the blob client this should probably commandline parameter, or potentially be part of the blob uri.

Reply:
This will be gone from the next commit.